src/components/test/QuickOpenModal.spec.js

```diff
@@ -210,6 +287,542 @@ describe("QuickOpenModal", () => {
+          searchType: "shortcuts",
+          symbols: {
+            functions: [{ title: "functionA" }, { title: "anonymous" }],
+            variables: {}
+          }
+        },
+        "shallow"
+      );
+      const id = "#";
+      wrapper.setState(() => ({
+        results: [{}, { id }],
+        selectedIndex: 1
+      }));
+      const event = {
+        key: "Enter"
+      };
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(props.selectLocation).not.toHaveBeenCalled();
+      expect(props.setQuickOpenQuery).toHaveBeenCalledWith(id);
+    });
+
+    it("on Tab, close modal", () => {
+      const { wrapper, props } = generateModal(
+        {
+          enabled: true,
+          query: ":34:12",
+          searchType: "goto"
+        },
+        "shallow"
+      );
+      const event = {
+        key: "Tab"
+      };
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(props.closeQuickOpen).toHaveBeenCalled();
+      expect(props.selectLocation).not.toHaveBeenCalled();
+    });
+
+    it("on ArrowUp, traverse results up with functions", () => {
+      const sourceId = "sourceId";
+      const { wrapper, props } = generateModal(
+        {
+          enabled: true,
+          query: "test",
+          searchType: "functions",
+          selectedSource: I.Map({ id: sourceId }),
+          symbols: {
+            functions: [{ title: "functionA" }, { title: "anonymous" }],
+            variables: {}
+          }
+        },
+        "shallow"
+      );
+      const event = {
+        preventDefault: jest.fn(),
+        key: "ArrowUp"
+      };
+      const location = {
+        start: {
+          line: 1
+        },
+        end: {
+          line: 3
+        }
+      };
+      wrapper.setState(() => ({
+        results: [{ id: "0", location }, { id: "1" }, { id: "2" }],
+        selectedIndex: 1
+      }));
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(event.preventDefault).toHaveBeenCalled();
+      expect(wrapper.state().selectedIndex).toEqual(0);
+      expect(props.highlightLineRange).toHaveBeenCalledWith({
+        end: 3,
+        sourceId: sourceId,
+        start: 1
+      });
+    });
+
+    it("on ArrowDown, traverse down with variables", () => {
+      const sourceId = "sourceId";
+      const { wrapper, props } = generateModal(
+        {
+          enabled: true,
+          query: "test",
+          searchType: "variables",
+          selectedSource: I.Map({ id: sourceId }),
+          symbols: {
+            functions: [{ title: "functionA" }, { title: "anonymous" }],
+            variables: {}
+          }
+        },
+        "shallow"
+      );
+      const event = {
+        preventDefault: jest.fn(),
+        key: "ArrowDown"
+      };
+      const location = {
+        start: {
+          line: 7
+        }
+      };
+      wrapper.setState(() => ({
+        results: [{ id: "0", location }, { id: "1" }, { id: "2" }],
```

Comment:
are these distinct for variables somehow? if not do we need to distinguish between these two?

Reply:
I think the code expects all variables to have a location, I just only give `location` to the one it expects to match. will throw an error on any of the others.

Reply:
ah sorry i wasnt too clear with my question: is there a difference between what happens when we traverse functions and variables?

Reply:
traversal is the same, but selection is different. there's other traversal tests looking at functions